misc/ngraph.1
.nr N -1
.nr D 5
.TH NGRAPH 1 Local
.UC 4
.SH NAME
Ngraph \- creating scientific 2-dimensional graphs
.SH SYNOPSIS
.B ngraph
[options] [data file(s) or ngp file]
.PP
.SH DESCRIPTION
.\" This defines appropriate quote strings for nroff and troff
.ds lq \&"
.ds rq \&"
.if t .ds lq ``
.if t .ds rq ''
.\" Just in case these number registers aren't set yet...
.if \nN==0 .nr N 10
.if \nD==0 .nr D 5
.I
\fINgraph\fR is the program to create scientific 2-dimensional graphs
for researchers and engineers. Graphs can be exported to PostScript,
SVG, PNG or PDF format.
.PP
.SH OPTIONS
Command line options are:
.PP
.TP
.B -L file
load ngp file
.TP
.B -x column
data column for the x axis
.TP
.B -y column
data column for the y axis
.TP
.B -X
use X axis as the x axis
.TP
.B -Y
use Y axis as the y axis
.TP
.B -U
use U axis as the x axis
.TP
.B -R
use R axis as the y axis
.TP
.B -d {mark | line | polygon | curve | diagonal | arrow | rectangle | rectangle_fill | rectangle_solid_fill | errorbar_x | errorbar_y | staircase_x | staircase_y | bar_x | bar_y | bar_fill_x | bar_fill_y | bar_solid_fill_x | bar_solid_fill_y | fit}
specify plot type
.TP
.B -m type
specify mark type (0-89)
.TP
.B -o size
specify mark size
.TP
.B -l style
specify line style
.TP
.B -w width
specify line width
.TP
.B -CR n
specify red component of the 1st plot color (0-255)
.TP
.B -CG n
specify green component of the 1st plot color (0-255)
.TP
.B -CB n
specify blue component of the 1st plot color (0-255)
.TP
.B -cr n
specify red component of the 2nd plot color (0-255)
.TP
.B -cg n
specify green component of the 2nd plot color (0-255)
.TP
.B -cb n
specify blue component of the 2nd plot color (0-255)
.TP
.B -s line
specify number of the head skip
.TP
.B -r step
specify number of the read step
.TP
.B -f line
specify number of the final line
.TP
.B -vx n
neighbor average of the x data
.TP
.B -vy n
neighbor average of the y data
.TP
.B -mx formula
specify math transformation for the x data
.TP
.B -my formula
specify math transformation for the y data
.TP
.B -ex {linear | log | inverse | MJD}
specify scale type of the x axis
.TP
.B -ey {linear | log | inverse | MJD}
specify scale type of the y axis
.TP
.B -minx value
specify minimum value of x axis
.TP
.B -maxx value
specify maximum value of x axis
.TP
.B -incx value
specify increment value of x axis
.TP
.B -miny value
specify minimum value of y axis
.TP
.B -maxy value
specify maximum value of y axis
.TP
.B -incy value
specify increment value of y axis
.TP
.B -g
load settings from data file
.TP
.B -png file
convert ngp file to PNG format
.TP
.B -pdf file
convert ngp file to PDF format
.TP
.B -ps file
convert ngp file to PostScript format
.TP
.B -eps file
convert ngp file to Encapsulated PostScript format
.TP
.B -svg file
convert ngp file to SVG format
.TP
.B -gra file
convert ngp file to GRA format
.TP
.B -dialog
show print dialog (with \-p option)
.TP
.B -p file
print ngp file
.TP
.B -n
output ngp-file to stdout
.TP
.B -h, --help
show help message
.TP
.B -v, --version
show version of \fINgraph\fR
.TP
.B -V, --VERSION
show detail information about \fINgraph\fR
.SH FILES
.TP
.I ${HOME}/.Ngraph/Ngraph.nsc
start up script
.TP
.I ${HOME}/.Ngraph/Ngraph.ini
configuration file
.TP
.I ${HOME}/.Ngraph/Ngraph.ngp
default graph
.SH AUTHORS
\fBSatoshi ISHIZAKA\fR
.PP
\fBHiroyuki Ito\fR <ZXB01226@nifty.com>

.SH "SEE ALSO"
.BR ngp2 (1)

.PP
.SH "COPYRIGHT"

.br

Copyright \(co 2003 Satoshi ISHIZAKA

.br

Copyright \(co 2008-2011 Hiroyuki Ito


NEWS
ngraph-gtk 6.03.48.gtk-26  2009/06/07
ngraph-gtk 6.03.48.gtk-25  2009/05/23
ngraph-gtk 6.03.48.gtk-24  2009/04/26
ngraph-gtk 6.03.48.gtk-23  2009/04/11
ngraph-gtk 6.03.47.gtk-22  2009/03/29
ngraph-gtk 6.03.47.gtk-21  2009/03/21
ngraph-gtk 6.03.47.gtk-20  2009/03/07
ngraph-gtk 6.03.47.gtk-19  2009/02/21
ngraph-gtk 6.03.47.gtk-18  2009/01/24
ngraph-gtk 6.03.47.gtk-17  2009/01/17
ngraph-gtk 6.03.47.gtk-16  2009/01/10
ngraph-gtk 6.03.47.gtk-15  2008/12/27
ngraph-gtk 6.03.47.gtk-14  2008/12/20
ngraph-gtk 6.03.47.gtk-13  2008/12/06
ngraph-gtk 6.03.47.gtk-12  2008/11/22
ngraph-gtk 6.03.47.gtk-11  2008/11/16
ngraph-gtk 6.03.47.gtk-10  2008/11/15
ngraph-gtk 6.03.47.gtk-09  2008/11/08
ngraph-gtk 6.03.47.gtk-08  2008/10/12
ngraph-gtk 6.03.47.gtk-07  2008/09/14
ngraph-gtk 6.03.47.gtk-06  2008/08/31
ngraph-gtk 6.03.47.gtk-05  2008/08/17
ngraph-gtk 6.03.30.gtk-04  2008/08/04
ngraph-gtk 6.03.30.gtk-03  2008/07/13
ngraph-gtk 6.03.30.gtk-02  2008/06/28
ngraph-gtk 6.03.30.gtk-01  2008/06/23
ngraph-gtk initial version 2008/06/08

misc/ngraph.bash
# ngraph completion 
# put this file in /etc/bash_completion.d/ 

have ngraph &&
_ngraph()
{
    local cur

    COMPREPLY=()
    cur=${COMP_WORDS[COMP_CWORD]}
    prev=${COMP_WORDS[COMP_CWORD-1]}

    case "$prev" in
        -png|-pdf|-ps|-eps|-svg|-gra|-p|-L)
            _filedir 'ngp'
	    return 0
            ;;
	-ex|-ey)
            COMPREPLY=( $( compgen -W 'linear log inverse MJD' -- $cur ) ) 
	    return 0
            ;;
	-d)
	    COMPREPLY=( $( compgen -W 'mark line polygon curve diagonal \
		arrow rectangle rectangle_fill rectangle_solid_fill \
		errorbar_x errorbar_y staircase_x staircase_y bar_x bar_y \
		bar_fill_x bar_fill_y bar_solid_fill_x bar_solid_fill_y fit' \
		-- $cur ) )
	    return 0
	    ;;
	-m)
	    COMPREPLY=( $( compgen -W "`seq 0 89`" -- $cur ) )
	    return 0
	    ;;
	-x|-y|-o|-l|-w|-cr|-cg|-cb|-CR|-CG|-CB|-s|-r|-f|-vx|-vy|-mx|-my|-minx|-maxx|-incx|-miny|-maxy|-incy)
	    return 0
	    ;;
    esac

    case "$cur" in
        -*)
            COMPREPLY=( $( compgen -W '-h --help --version --VERSION \
		-L -x -y -X -Y -d -m -o -l -w -cr -cg -cb -CR -CG -CB -s -r -f \
		-vx -vy -mx -my -ex -ey -minx -maxx -incx -miny -maxy -incy -g \
		-png -pdf -ps -eps -svg -gra -n -p -dialog' -- $cur ) ) 
            ;;
	*)
	    _filedir
	    ;;
    esac
}
[ "${have:-}" ] && complete -F _ngraph $filenames ngraph


have ngp2 &&
_ngp2()
{
    local cur

    COMPREPLY=()
    cur=${COMP_WORDS[COMP_CWORD]}
    case "$cur" in
        -*)
            COMPREPLY=( $( compgen -W '-h --help -I -a -A -c \
		-ps -ps3 -ps2 -eps -eps3 -eps2 \
		-wmf -pdf -svg -svg1.1 -svg1.2 -png' -- $cur ) ) 
            ;;
	*)
	    _filedir 'ngp'
	    ;;
    esac
}
[ "${have:-}" ] && complete -F _ngp2 $filenames ngp2


# have gra2ps &&
# _gra2ps()
# {
#     local cur

#     COMPREPLY=()
#     cur=${COMP_WORDS[COMP_CWORD]}
#     prev=${COMP_WORDS[COMP_CWORD-1]}

#     case "$prev" in
#         -o)
#             _filedir '@(ps|eps)'
# 	    return 0
#             ;;
# 	-i)
#             _filedir 'ps'
# 	    return 0
#             ;;
# 	-p)
# 	    COMPREPLY=( $( compgen -W 'a3 a4 b4 a5 b5 letter legal' -- $cur ) )
# 	    return 0
# 	    ;;
# 	-s)
# 	    return 0
# 	    ;;
#     esac

#     case "$cur" in
#         -*)
#             COMPREPLY=( $( compgen -W '-o -i -c -e -p -l -r' -- $cur ) ) 
#             ;;
# 	*)
# 	    _filedir 'gra'
# 	    ;;
#     esac
# }
# [ "${have:-}" ] && complete -F _gra2ps $filenames gra2ps

# have gra2wmf &&
# _gra2wmf()
# {
#     local cur

#     COMPREPLY=()
#     cur=${COMP_WORDS[COMP_CWORD]}
#     prev=${COMP_WORDS[COMP_CWORD-1]}

#     case "$prev" in
#         -o)
#             _filedir 'wmf'
# 	    return 0
#             ;;
# 	-d)
# 	    return 0
# 	    ;;
#     esac

#     case "$cur" in
#         -*)
#             COMPREPLY=( $( compgen -W '-o -d' -- $cur ) ) 
#             ;;
# 	*)
# 	    _filedir 'gra'
# 	    ;;
#     esac
# }
# [ "${have:-}" ] && complete -F _gra2wmf $filenames gra2wmf
